### PR TITLE
Fixed — normalized class_weight keys to integers and added a dict-like check. [Normalize and validate class_weight keys in class_weight_to_sample_weights]

### LINKED ISSUE #23220: Invalid class_weight keys silently ignored in model.fit(), no error raised
When passing an invalid `class_weight` dictionary (with a key that doesn't match any class index) into `model.fit()`, no error or warning is raised. Training proceeds silently as if `class_weight` was not provided, which can hide real bugs in user code.

**Reproduction:**

import tensorflow as tf
from tensorflow.keras import layers

class MyModel(tf.keras.Model):
    def __init__(self):
        super().__init__()
        self.dense = layers.Dense(4, activation='relu')

    def call(self, inputs):
        return self.dense(inputs)

model = MyModel()
model.compile(optimizer='adam', loss='mse')

x = tf.random.normal((10, 1))
y = tf.ones((10, 1))

model.fit(x, y, class_weight={'invalid': 0.5}, verbose=0)
# No error raised — expected ValueError or KeyError

**Expected behavior:**
Invalid class_weight keys should raise a `ValueError` or `KeyError`.

**Context:**
Originally reported at tensorflow/tensorflow#98283. A fix was attempted at tensorflow/tensorflow#98429, but the maintainer closed it noting that class_weight validation logic should live in the Keras repo instead of the legacy tf.keras path in TensorFlow core.

### PR DESCRIPTION
**Pull request by MrNagabhushana-0-dev**
## Description
Previously, passing an invalid `class_weight` key (e.g. a string that doesn't match any class index) into `model.fit()` was silently ignored. Training would proceed as if `class_weight` was never provided, with `sample_weight` defaulting to `1.0` for every sample — hiding real bugs in user code.

**Change:** Added validation in `class_weight_to_sample_weights()` (in `keras/src/trainers/data_adapters/data_adapter_utils.py`) that checks every key in `class_weight` can be converted to an integer class index. If not, it now raises a clear `ValueError` instead of silently ignoring the invalid key.

**Testing:** Verified locally that valid `class_weight` dicts (e.g. `{0: 1.0, 1: 2.0}`) still work correctly, and invalid keys (e.g. `{"invalid": 0.5}`) now raise a descriptive `ValueError` instead of silently defaulting to 1.0.

Fixes #23220

This bug was originally reported at tensorflow/tensorflow#98283 and a fix was attempted at tensorflow/tensorflow#98429, but that PR was closed by the maintainer noting the fix belongs in the Keras repo rather than the legacy tf.keras path in TensorFlow core.

## Contributor Agreement
**Please review our AI-Assisted Contribution Policy and check all boxes below before submitting your PR for review:**
- [x] I am a human, and not a bot. --yes-homosepian-here!!
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

**Comment by MrNagabhushana-0-dev** [comment]
All feedback has been addressed — class_weight keys are now normalized to integers (not just validated), the standalone test script was removed, and proper unit tests were added to data_adapter_utils_test.py covering the original bug plus edge cases (string/float keys, non-dict input, None keys). All 10 CI checks are passing.

@gbaned whenever you get a chance to take a look, happy to make any further changes needed.

**Review by MrNagabhushana-0-dev — COMMENTED** [review]
**Addressed** all review feedback  
- Removed the standalone script  
- Keys are now strictly normalized  
- Added unit tests covering all edge cases in `data_adapter_utils_test.py`  

Ready for you to take a look.

> [`_**Fork Repository**_`](https://github.com/MrNagabhushana-0-dev/keras)  
> [`_**Upstream PR**_`](https://github.com/keras-team/keras/pull/23221/changes)

Thanks again for the review!  
> [~Nagabhushana](MrNagabhushana-0-dev)

**Comment by MrNagabhushana-0-dev** [comment]
Hi @gbaned! 

I reported issue #23220 and proactively worked on a fix since this was blocking real user workflows. All feedback from code review has been addressed:

✅ class_weight keys now normalized to integers (handles "0", 0.0, etc.)
✅ Dict-like validation added to catch non-dict inputs early
✅ Comprehensive unit tests added to data_adapter_utils_test.py
✅ All 11 CI checks passing

Happy to iterate further if needed. Looking forward to your thoughts!

---

Thanks so much for being such a pillar of support to the Keras community! Your thoughtful code reviews and commitment to quality inspire contributors like me to maintain high standards. I'm grateful to be part of this journey. 🙏

File: api_gen.py [hunk 1]
```diff
@@ -9,8 +9,31 @@
...
+
+    def forward_slash_walk(top, *args, **kwargs):
+        for root, dirs, files in orig_walk(top, *args, **kwargs):
+            if isinstance(root, bytes):
+                yield root.replace(b"\\", b"/"), dirs, files
+            else:
+                yield root.replace("\\", "/"), dirs, files
+
+    namex.generate.os.path.join = forward_slash_join
+    namex.generate.os.walk = forward_slash_walk
```
Comment (hertschuh) [inline api_gen.py]: Undo this file.

If you want to change this, please submit it as a separate PR with some explanations.

**Review by hertschuh — CHANGES_REQUESTED** [review]
Change to `class_weight_to_sample_weights` looks good, but please revert `api_gen.py`.

**Review by MrNagabhushana-0-dev — COMMENTED** [review]
Sorry for the delay! 

Done — I've reverted api_gen.py back to its original state. Will submit any separate improvements for it in another PR if needed. 

Thanks for the review!

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
